### PR TITLE
use the user set usage in vertexbuffer

### PR DESCRIPTION
**Pull request by juakob**
implemented just in opengl 
the rest are just updated so they compile

### CODE REVIEW

File: Backends/Graphics4/Direct3D9/Sources/Kore/VertexBufferImpl.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@ Graphics4::VertexBuffer* VertexBufferImpl::_current = nullptr;
 
 VertexBufferImpl::VertexBufferImpl(int count, int instanceDataStepRate) : myCount(count), instanceDataStepRate(instanceDataStepRate) {}
 
-Graphics4::VertexBuffer::VertexBuffer(int count, const VertexStructure& structure, int instanceDataStepRate) : VertexBufferImpl(count, instanceDataStepRate) {
+Graphics4::VertexBuffer::VertexBuffer(int count, const VertexStructure& structure, usage: Usage, int instanceDataStepRate) : VertexBufferImpl(count, instanceDataStepRate) {
```
Comment (sh-dave) [inline Backends/Graphics4/Direct3D9/Sources/Kore/VertexBufferImpl.cpp]: `usage: Usage` should probably be `Usage usage` as we're not in haxe land anymore?

File: Backends/Graphics3/OpenGL1/Sources/Kore/VertexBufferImpl.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@ VertexBufferImpl::VertexBufferImpl(int count, int instanceDataStepRate) : myCoun
 #endif
 }
 
-Graphics3::VertexBuffer::VertexBuffer(int vertexCount, const Graphics4::VertexStructure& structure, int instanceDataStepRate)
+Graphics3::VertexBuffer::VertexBuffer(int vertexCount, const Graphics4::VertexStructure& structure, usage: Usage, int instanceDataStepRate)
```
Comment (sh-dave) [inline Backends/Graphics3/OpenGL1/Sources/Kore/VertexBufferImpl.cpp]: `usage: Usage` should probably be `Usage usage` as we're not in haxe land anymore?

**Comment by juakob** [comment]
thanks dave! my bad, this is why you shouldn't open a request before you go to sleep XD.

Cant make direct3D11 work using dynamic as a type

**Review by juakob — COMMENTED** [review]
not working direct3d11 implementation

**Comment by juakob** [comment]
Ok I think now its ready, still sombody should look at the direct3d11 vertexbuffer class https://github.com/Kode/Kore/pull/268/files#diff-834d90cf58e85cd04750c5bfdd3b63b3
It does work, but I don't know much c++ so I might be doing somthing horribly wrong, especially worry about the memory/pointers stuff. It was implemented using this guide https://msdn.microsoft.com/en-us/library/windows/desktop/dn508285(v=vs.85).aspx

**Pull request merged**